test/unit/table-detection.spec.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2019 AXA Group Operations S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import * as fs from 'fs';
+import { withData } from 'leche';
+import 'mocha';
+import { TableDetectionModule } from '../../server/src/processing/TableDetectionModule/TableDetectionModule';
+import {
+	Document,
+	Table,
+	TableCell,
+	TableRow,
+} from '../../server/src/types/DocumentRepresentation';
+import { json2document } from '../../server/src/utils/json2document';
+
+import { runModules } from './../helpers';
+
+describe('Table Detection Module', () => {
+	withData(
+		{
+			'simple table': ['table-detection.json', 1, 6, 31],
+			'table inside document': ['foo.json', 1, 7, 43],
+		},
+		(fileName, tableCount, rowCount, cellCount) => {
+			let docBefore: Document;
+			let docAfter: Document;
+
+			before(done => {
+				const json = JSON.parse(
+					fs.readFileSync(__dirname + '/assets/' + fileName, { encoding: 'utf8' }),
+				);
+
+				docBefore = json2document(json);
+				docBefore.inputFile = __dirname + '/assets/' + fileName.replace('.json', '.pdf');
+
+				runModules(docBefore, [new TableDetectionModule()]).then(after => {
```

Comment:
This is executing real Table detection script instead of mock table data, this makes test to be slow. 

Why not use 'TableExtractorStub' object as module dependency injection ??

`const extractorOutput = fs.readFileSync(
		`${__dirname}/assets/mocks/table-detection-one-table.json`,
		'utf8',
	);
	const oneTableDetectedExtractor = new TableExtractorStub(0, '', extractorOutput);
.....
runModules(d, [new TableDetectionModule(null, oneTableDetectedExtractor)])
`

Reply:
as this was actually testing the camelot library, and the way proposed by @jvalls-axa is already working in `test/table-detection.spec.ts`, this PR is not adding any value to the project. closed!